packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -153,16 +208,12 @@ export class PersonalizeMiddleware extends MiddlewareBase {
       );
       return response;
     }
-
-    const site = this.getSite(req, res);
```

Comment:
`getSite`, `initializeEngageServer` and especially `handleCookie` should all occur _after_ skip conditions above.